app/src/main/java/is/xyz/mpv/MPVActivity.kt

```diff
@@ -363,6 +371,9 @@ class MPVActivity : AppCompatActivity(), MPVLib.EventObserver, TouchGesturesObse
             moveTaskToBack(true)
         } else {
             MPVLib.command(arrayOf("loadfile", filepath))
+            // When not appending to a playlist, there's no reason to keep old files open anymore
+            previousPfds.forEach { it.close() }
+            openPfds.removeAll(previousPfds)
```

Comment:
mpv commands are async so it might still be using the fd when this runs.
only safe thing to do it keep them...

Reply:
Ah shoot, I didn't realize that.

Reply:
Oh, a custom protocol handler sounds like a great idea. I didn't realize mpv could do this without changing its core code, but it looks like `mpv/stream_cb.h` is exactly for this purpose. I'll try implementing this approach this weekend.

Reply:
I've opened https://github.com/mpv-android/mpv-android/pull/1090. It turns out ffmpeg 7 natively supports `content://` URIs as long as it has access to the application context, so there was no need to implement a custom protocol handler.